### ISSUE TITLE
cacheDataProviderProxy not called with ra-data-graphql-simple

### ISSUE THREAD

**Issue by RWOverdijk** [marmelab/react-admin#6438]
I wasn't sure if I should put this here or on the ra-data-graphql-simple repo. I'm not even sure if this is an actual bug or just me doing something wrong.

**What you were expecting:**

I was expecting cacheDataProviderProxy's `get()` to be called when used with graphql. I notice that every Edit performs a get and I want to use the cacheDataProviderProxy to prevent that.

**What happened instead:**

I've added several logs and none of them are being called. I verified my logs are there in the browser, and they are.

<details>
  <summary>Screenshot collapsed, click to expand.</summary>
  
![image](https://user-images.githubusercontent.com/781745/125818027-7771a437-a355-4b0e-a461-ee809dd4f506.png)
</details>

**Steps to reproduce:**

Using the default code in the ra-data-graphql-simple readme with the addition of the cacheDataProviderProxy.

<details>
  <summary>Code collapsed, click to expand.</summary>
  
```jsx
import * as React from 'react';
import { Component } from 'react';
import buildGraphQLProvider from 'ra-data-graphql-simple';
import { Admin, cacheDataProviderProxy } from 'react-admin';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = { dataProvider: null };
  }

  componentDidMount() {
    // 👇 This is where I added the `cacheDataProviderProxy()` 👇 
    buildGraphQLProvider({ clientOptions: { uri: 'http://localhost:4000/graphql' }})
        .then(dataProvider => this.setState({ dataProvider: cacheDataProviderProxy(dataProvider) }));
  }

  render() {
    const { dataProvider } = this.state;

    if (!dataProvider) {
      return <div>Loading</div>;
    }

    return (
        <Admin dataProvider={dataProvider}>
        </Admin>
    );
  }
}

export default App;
```
</details>

**Environment**

* React-admin version: 3.16.6 (waiting for material-ui fix)
* Last version that did not exhibit the issue (if applicable): N/A
* React version: 17.0.2
* Browser: Chrome
* Stack trace (in case of a JS error): N/A

**Comment by djhi** [comment]
Hi, and thanks for your question. As explained in [the react-admin contributing guide](https://github.com/marmelab/react-admin/blob/master/.github/CONTRIBUTING.md), we use the GitHub issues for bugs and feature requests only. 

For support question ("How To", usage advice, or troubleshooting your own code), you have three options:

- **Community support**: Ask your question on [StackOverFlow](https://stackoverflow.com/questions/tagged/react-admin). This makes your question easy to find by the developer community. StackOverFlow has great SEO, gamification, voting, and reputation system.
- **Community support**: Ask your question on [Discord](https://discord.gg/MR9QFxMc). Some react-admin users and contributors might have time to answer your questions.
- **Professional support**: If you're a [React-Admin Enterprise Edition](https://marmelab.com/ra-enterprise/#support) subscriber, you have unlimited access to the Professional Support service. This service guarantees that you'll get an answer shortly.

If your problem is urgent, please consider subscribing to the [Enterprise Edition](https://marmelab.com/ra-enterprise/), starting at 125€. It's a good way to contribute to the sustainability of react-admin. 

So I'm closing this issue, and inviting you to ask your question in one of the support channels above.

**Comment by djhi** [comment]
Sorry I closed that too fast. This is because the graphql dataproviders have not been updated to the new dataProvider format. They still return a single function instead of an object. Flagging this as an enhancement

**Comment by RWOverdijk** [comment]
Thanks.

If nobody picks this up I'll see if I can tackle this next time I work on this project. How hard can it be, right. Sounds like the function is replaced by an object literal where the previous action argument is now the key. Is that correct?

**Comment by djhi** [comment]
Exactly!

**Comment by larsivi** [comment]
Maybe I am hitchhiking on an issue I shouldn't, but after an upgrade I got an issue with my graphql requests that I'm struggling to see why happens. Our queries have more data than what is available to send with the mutations (because these are generated fields on the database side). What I used to do, was to filter these 1) in transform in the edit forms 2) filter in our dataprovider implementation (that is just a wrapper around ra-data-graphql-prisma that uses ra-data-graphql). But my logging suggests that this filtering works, but that the fields somehow still are added to requests, causing havoc server side. Could this be because the ra-data-graphql-prisma needs to be updated to the object format?
